ngi_reports/utils/entities.py

```diff
@@ -0,0 +1,453 @@
+                                        'ApplicationVersion': fc_runp.get('MCSVersion')
+                                        }
+            elif fcObj.type == 'NextSeq500':
+                fcObj.seq_software = {'RTAVersion': fc_runp.get('RTAVersion', fc_runp.get('RtaVersion')),
+                                        'ApplicationName': fc_runp.get('ApplicationName', fc_runp.get('Application', fc_runp.get('Setup').get('ApplicationName'))),
+                                        'ApplicationVersion': fc_runp.get('ApplicationVersion', fc_runp.get('Setup').get('ApplicationVersion'))
+                                        }
+            else:
+                fcObj.seq_software = {'RTAVersion': fc_runp.get('RTAVersion', fc_runp.get('RtaVersion')),
+                                        'ApplicationName': fc_runp.get('ApplicationName', fc_runp.get('Application')),
+                                        'ApplicationVersion': fc_runp.get('ApplicationVersion')
+                                        }
+
+            ## Collect quality info for samples and collect lanes of interest
+            for stat in fc_details.get('illumina',{}).get('Demultiplex_Stats',{}).get('Barcode_lane_statistics',[]):
+                try:
+                    if re.sub('_+','.',stat['Project'],1) != self.ngi_name and stat['Project'] != self.ngi_name:
+                        continue
+
+                    lane = stat['Lane']
+                    if fc['db'] == 'x_flowcells':
+                        sample = stat['Sample']
+                        barcode = stat['Barcode sequence']
+                        qval_key, base_key = ('% >= Q30bases', 'PF Clusters')
+
+                    else:
+                        sample = stat['Sample ID']
+                        barcode = stat['Index']
+                        qval_key, base_key = ('% of >= Q30 Bases (PF)', '# Reads')
+
+                    if kwargs.get('samples', []) and sample not in kwargs.get('samples', []):
+                        continue
+
+                    try:
+                        r_idx = '{}_{}_{}'.format(lane, fcObj.name, barcode)
+                        r_len_list = [x['NumCycles'] for x in fcObj.run_setup if x['IsIndexedRead'] == 'N']
+                        r_len_list = [int(x) for x in r_len_list]
+                        r_num = len(r_len_list)
+                        qval = float(stat.get(qval_key))
+                        pfrd = int(stat.get(base_key).replace(',',''))
+                        pfrd = pfrd/2 if fc['db'] == 'flowcell' else pfrd
+                        base = pfrd * sum(r_len_list)
+                        sample_qval[sample][r_idx] = {'qval': qval, 'reads': pfrd, 'bases': base}
+
+                    except (TypeError, ValueError, AttributeError) as e:
+                        log.warn('Something went wrong while fetching Q30 for sample {} with barcode {} in FC {} at lane {}'.format(sample, barcode, fcObj.name, lane))
+                        pass
```

Comment:
I'm not a fan of this general try-except-pass section. I guess we don't really know what's going wrong and it seems like we don't care?